File: arrays/meeting_rooms.java
// https://leetcode.com/problems/meeting-rooms/

// Sort the array by starting time
// Check for overlapping, i.e. finishing time of i > starting time of i+1

class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[0]));
        for (int i = 0; i < intervals.length-1; i++) {
            if (intervals[i][1] > intervals[i+1][0]) {
                return false;
            }
        }
        return true;
    }
}

File: arrays/meeting-rooms.py
# https://leetcode.com/problems/meeting-rooms/
from typing import List
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        ans = True
        for i in range(len(intervals)):
            for j in range(len(intervals)):
                if intervals[i][0] <= intervals[j][0] < intervals[i][1]:
                    ans = False
        return ans

test = Solution()
time = [[7,10],[2,4]]
# time = [[0,30],[5,10],[15,20]]
print(test.canAttendMeetings(time))
